hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/FinalizeJoinOp.java

```diff
@@ -188,6 +194,7 @@ protected void writeInternalImpl(ObjectDataOutput out) throws IOException {
         out.writeObject(preJoinOp);
         out.writeObject(postJoinOp);
         out.writeBoolean(deferPartitionProcessing);
+        out.writeByte(clusterTopologyIntent.ordinal());
```

Comment:
minor: giving an ID to each clusterTopologyIntent types could be more safer than using ordinal

Reply:
done [here](https://github.com/hazelcast/hazelcast/pull/21844/files#diff-01306ce7cd60d256f4dd0a184b8f948253dec322c461d4cb8ec1a3239826ea46R74)